File: mdds-dto/src/main/java/com/mdds/domain/ArtifactFormat.java
/*
 * Copyright (c) 2025 Oleksiy Oleksandrovych Sayankin. All Rights Reserved.
 * Refer to the LICENSE file in the root directory for full license details.
 */
package com.mdds.domain;

import java.util.Arrays;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

/** Supported file formats for input/output artifacts. */
@RequiredArgsConstructor
public enum ArtifactFormat {
  CSV("csv"),
  JSON("json");

  @Getter private final String value;

  public static ArtifactFormat from(String raw) {
    if (raw == null || raw.isBlank()) {
      throw new UnknownArtifactFormatException("Artifact format must not be null or blank.");
    }
    return Arrays.stream(values())
        .filter(t -> t.value.equalsIgnoreCase(raw.trim()))
        .findFirst()
        .orElseThrow(
            () ->
                new UnknownArtifactFormatException(
                    "Unknown or unsupported artifact format: '" + raw + "'."));
  }
}

File: mdds-python-worker-runtime/src/main/python/mdds_worker_runtime/domain/artifact_format.py
# Copyright (c) 2025 Oleksiy Oleksandrovych Sayankin. All Rights Reserved.
# Refer to the LICENSE file in the root directory for full license details.

from enum import Enum


class UnknownArtifactFormatError(ValueError):
    pass


class ArtifactFormat(Enum):
    """Artifact format. This class is similar to the Java class."""

    CSV = "csv"
    JSON = "json"

    @classmethod
    def from_raw(cls, raw: str | None) -> "ArtifactFormat":
        if raw is None or raw.strip() == "":
            raise UnknownArtifactFormatError(
                "Artifact format must not be null or blank."
            )

        normalized = raw.strip().lower()
        for item in cls:
            if item.value == normalized:
                return item

        raise UnknownArtifactFormatError(
            f"Unknown or unsupported artifact format: '{raw}'."
        )
